src/host/ft_host/CJK_DbcsTests.cpp

```diff
@@ -5,11 +5,33 @@
 #include <io.h>
 #include <fcntl.h>
 #include <iostream>
-#include <iomanip>
 
-#define ENGLISH_US_CP 437u
 #define JAPANESE_CP 932u
 
+// CHAR_INFO's .Char member is a union of a wchar_t UnicodeChar and char AsciiChar.
+// If they share the same offsetof we can write the lower byte of the former to
+// overwrite the latter, while ensuring that the high byte is properly cleared to 0.
+static_assert(offsetof(CHAR_INFO, Char.UnicodeChar) == offsetof(CHAR_INFO, Char.AsciiChar));
+
+template<typename T>
+constexpr CHAR_INFO makeCharInfo(T ch, WORD attr)
+{
+    CHAR_INFO info{};
+    // If T is a char, it'll be a signed integer, whereas UnicodeChar is an unsigned one.
+    // A negative char like -1 would then result in a wchar_t of 0xffff instead of the expected 0xff.
+    // Casting ch to a unsigned integer first prevents such "sign extension".
+    info.Char.UnicodeChar = static_cast<WCHAR>(til::as_unsigned(ch));

```

Comment:
but why does it support `T` instead of just being WCHAR? Are there legitimate cases where we want to pass a char, and stuff it into the `WCHAR` field??

Reply:
Initially I had an overload for `char` and one for `wchar_t`.
But if you pass a `0x1234` it'll complain that it doesn't know which overload you meant. I could've used a `if constexpr`, but then I'd have to add an static assert in case it's neither of the two types, etc. This code works as expected and is a direct copy of my upcoming PR where the same code needs to exist in `directio.cpp` (it writes `char`s to `UnicodeChar` so that the high byte is cleared).

Reply:
I should mention that I thought this would be super easy to write actually. Like most people I think of `char`s as unsigned bytes and so I just wrote this:
```cpp
info.Char.UnicodeChar = ch;
```

I have to confess that I was a bit angry that this didn't work as expected and that I didn't want to bend to the compiler there.